src/structs.py

```diff
@@ -1020,6 +1022,36 @@ def __str__(self) -> str:
         return repr(self)
 
 
+@dataclass(frozen=True, eq=False, repr=False)
+class InteractionRequest:
+    """A request for interacting with a training task during online learning.
+    Contains the index for that training task, an acting policy, a query policy,
+    and a termination function.
+
+    Note: the act_policy will not be called on the state where the
```

Comment:
@tomsilver a bit strange but the only thing i could think of that makes sense, esp. since we'll want to allow queries about just the initial state without any actions.

Reply:
it doesn't seem that strange to me